### PR TITLE
[GOBBLIN-1812] Mockito should only be a test compile dependency

### PR DESCRIPTION
**Pull request by homatthew**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [ ] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-1812] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1812

There was an error in a previous commit that changed mockito-core dependency to mockito-inline. Mockito-inline fundamentally changes how mocking works, and there were mockito was a compile dependency that broke downstream dependencies. 

In general, since we are using the plugin approach (see https://github.com/apache/gobblin/pull/3651/files#r1164827929 for an example), there's no reason to use mockito-inline as our dependency. In mockito 5, mockito-inline becomes the default but since we are still on java 8, we should just cross that bridge when we get there

Check below to see how the buildscan now no longer has mockito in compile path. Only test compile path

[Buildscan](https://scans.gradle.com/s/wtnou474poohi/dependencies?dependencies=mockito&expandAll)
<details>
image

```
gobblin-api
testCompileClasspath0.006s
org.mockito:mockito-core:4.11.0
:gobblin-audit
testCompileClasspath0.010s
org.mockito:mockito-core:4.11.0
:gobblin-aws
testCompileClasspath0.180s
org.mockito:mockito-core:4.11.0
:gobblin-binary-management
testCompileClasspath0.018s
org.mockito:mockito-core:4.11.0
:gobblin-cluster
testCompileClasspath0.149s
org.mockito:mockito-core:4.11.0
:gobblin-compaction
testCompileClasspath0.067s
org.mockito:mockito-core:4.11.0
:gobblin-completeness
testCompileClasspath0.018s
org.mockito:mockito-core:4.11.0
:gobblin-config-management:gobblin-config-client
testCompileClasspath0.008s
org.mockito:mockito-core:4.11.0
:gobblin-config-management:gobblin-config-core
testCompileClasspath0.008s
org.mockito:mockito-core:4.11.0
:gobblin-core
testCompileClasspath0.029s
org.mockito:mockito-core:4.11.0
:gobblin-core-base
testCompile0.168s
org.mockito:mockito-core:4.11.0
testCompileClasspath0.049s
org.mockito:mockito-core:4.11.0
:gobblin-data-management
testCompileClasspath0.077s
org.mockito:mockito-core:4.11.0
:gobblin-iceberg
testCompileClasspath0.023s
org.mockito:mockito-core:4.11.0
:gobblin-metrics-libs:gobblin-metrics
testCompileClasspath0.008s
org.mockito:mockito-core:4.11.0
:gobblin-metrics-libs:gobblin-metrics-base
testCompileClasspath0.008s
org.mockito:mockito-core:4.11.0
:gobblin-modules:gobblin-azkaban
testCompileClasspath0.175s
org.mockito:mockito-core:4.11.0
:gobblin-modules:gobblin-compliance
testCompileClasspath0.116s
org.mockito:mockito-core:4.11.0
:gobblin-modules:gobblin-couchbase
testCompileClasspath0.158s
org.mockito:mockito-core:4.11.0
:gobblin-modules:gobblin-elasticsearch
testCompileClasspath0.263s
org.mockito:mockito-core:4.11.0
:gobblin-modules:gobblin-eventhub
testCompileClasspath0.036s
org.mockito:mockito-core:4.11.0
:gobblin-modules:gobblin-http
testCompileClasspath0.216s
org.mockito:mockito-core:4.11.0
:gobblin-modules:gobblin-kafka-08
testCompileClasspath0.123s
org.mockito:mockito-core:4.11.0
:gobblin-modules:gobblin-kafka-09
testCompileClasspath0.329s
org.mockito:mockito-core:4.11.0
:gobblin-modules:gobblin-kafka-1
testCompileClasspath0.201s
org.mockito:mockito-core:4.11.0
:gobblin-modules:gobblin-kafka-common
testCompile0.091s
org.mockito:mockito-core:4.11.0
testCompileClasspath0.026s
org.mockito:mockito-core:4.11.0
:gobblin-modules:gobblin-metrics-graphite
testCompileClasspath0.086s
org.mockito:mockito-core:4.11.0
:gobblin-modules:gobblin-metrics-hadoop
testCompileClasspath0.008s
org.mockito:mockito-core:4.11.0
:gobblin-modules:gobblin-metrics-influxdb
testCompileClasspath0.132s
org.mockito:mockito-core:4.11.0
:gobblin-modules:gobblin-orc
testCompileClasspath0.277s
org.mockito:mockito-core:4.11.0
:gobblin-modules:gobblin-parquet
testCompileClasspath0.128s
org.mockito:mockito-core:4.11.0
:gobblin-modules:gobblin-parquet-apache
testCompileClasspath0.073s
org.mockito:mockito-core:4.11.0
:gobblin-modules:gobblin-sql
testCompileClasspath0.142s
org.mockito:mockito-core:4.11.0
:gobblin-modules:gobblin-troubleshooter
testCompileClasspath0.104s
org.mockito:mockito-core:4.11.0
:gobblin-modules:google-ingestion
testCompileClasspath0.016s
org.mockito:mockito-core:4.11.0
:gobblin-restli:gobblin-flow-config-service:gobblin-flow-config-service-client
testCompileClasspath0.243s
org.mockito:mockito-core:4.11.0
:gobblin-restli:gobblin-throttling-service:gobblin-throttling-service-client
testCompileClasspath0.056s
org.mockito:mockito-core:4.11.0
:gobblin-restli:gobblin-throttling-service:gobblin-throttling-service-server
testCompileClasspath0.010s
org.mockito:mockito-core:4.11.0
:gobblin-runtime
testCompile0.211s
org.mockito:mockito-core:4.11.0
testCompileClasspath0.037s
org.mockito:mockito-core:4.11.0
:gobblin-runtime-hadoop
testCompileClasspath0.111s
org.mockito:mockito-core:4.11.0
:gobblin-service
testCompileClasspath0.117s
org.mockito:mockito-core:4.11.0
:gobblin-test-utils
testCompile0.068s
org.mockito:mockito-core:4.11.0
testCompileClasspath0.064s
org.mockito:mockito-core:4.11.0
:gobblin-utility
testCompileClasspath0.073s
org.mockito:mockito-core:4.11.0
:gobblin-yarn
testCompileClasspath0.183s
org.mockito:mockito-core:4.11.0
```

</details>

### Description
- [X] Here are some details about my PR, including screenshots (if applicable):

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-data-management/src/test/java/org/apache/gobblin/util/test/RetentionTestDataGenerator.java
Comment (homatthew) [inline gobblin-data-management/src/test/java/org/apache/gobblin/util/test/RetentionTestDataGenerator.java:1]: Test data generator should be in tests and not src

File: gobblin-data-management/src/test/java/org/apache/gobblin/util/test/RetentionTestHelper.java
Comment (homatthew) [inline gobblin-data-management/src/test/java/org/apache/gobblin/util/test/RetentionTestHelper.java:1]: Test helper should be in tests and not src

File: gobblin-data-management/src/test/java/org/apache/gobblin/util/test/RetentionTestDataGenerator.java
Comment (arjun4084346) [reply gobblin-data-management/src/test/java/org/apache/gobblin/util/test/RetentionTestDataGenerator.java:1]: these classes are used outside OSS gobblin.
if they are moved to test, they will not be exported in the jar, not at least till you do some gradle magic.

File: gobblin-data-management/src/test/java/org/apache/gobblin/util/test/RetentionTestHelper.java
Comment (arjun4084346) [reply gobblin-data-management/src/test/java/org/apache/gobblin/util/test/RetentionTestHelper.java:1]: same here

**Review by Will-Lo — APPROVED** [review]

File: gobblin-data-management/src/test/java/org/apache/gobblin/util/test/RetentionTestDataGenerator.java
Comment (Will-Lo) [reply gobblin-data-management/src/test/java/org/apache/gobblin/util/test/RetentionTestDataGenerator.java:1]: Would recommend not renaming classes if possible so downstream users aren't affected

**Comment by Will-Lo** [comment]
With some consideration if there are downstream projects using this test helper class it should follow this link to utilize it in their own test classes.
https://discuss.gradle.org/t/how-do-i-include-test-sources-along-side-test-jar/12380

File: gobblin-data-management/src/test/java/org/apache/gobblin/util/test/RetentionTestDataGenerator.java
Comment (homatthew) [reply gobblin-data-management/src/test/java/org/apache/gobblin/util/test/RetentionTestDataGenerator.java:1]: FWIW, internally, these test classes are not used at all. References to this are already donated to OSS and are redundant

**Pull request merged**